agentrs/src/conf.rs

```diff
@@ -0,0 +1,425 @@
+        let hoop_key = get_token();
+        if hoop_key.is_none() {
+            Err(anyhow::anyhow!(
+                "HOOP_KEY environment variable is not set. This may lead to insecure configurations."
+            ))?;
+        }
+        let ip_addresses = vec!["127.0.0.1".parse::<IpAddr>()?, "::1".parse::<IpAddr>()?];
+        let cert_key_pair = crate::x509::generate_gateway_cert(ip_addresses);
+        let tls = match cert_key_pair {
+            Ok(cert_key_pair) => {
+                let (certificates, private_key) = cert_key_pair.to_rustls();
+                let certificates = vec![certificates];
+                let cert_source = crate::tls::CertificateSource::External {
+                    certificates,
+                    private_key,
+                };
+                Some(Tls::init(cert_source, true)?)
+            }
+            Err(e) => {
+                error!("Failed to generate certificate: {e}");
+                None
+            }
+        };
+        if tls.is_none() {
+            Err(anyhow::anyhow!("TLS error generating certificates."))?;
+        }
+
+        let conf = Conf {
+            hostname: "gateway.hoop".to_string(),
+            tls,
+            token: hoop_key,
+        };
+
+        let config_handler = Self {
+            conf: Arc::new(conf),
+        };
+        Ok(config_handler)
+    }
+
+    pub fn init() -> anyhow::Result<Self> {
+        let hoop_auto_gencert = std::env::var("HOOP_CERT")
+            .unwrap_or_else(|_| "true".to_string())
+            .parse::<bool>()
+            .unwrap_or(true);
+        let conf_manager = match hoop_auto_gencert {
+            true => Self::auto_gen_conf(),
+            false => Self::init_from_files(),
+        }?;
+        return Ok(conf_manager);
+    }
+
+    pub fn init_from_files() -> anyhow::Result<Self> {
+        let hoop_key = get_token();
+        if hoop_key.is_none() {
+            Err(anyhow::anyhow!(
+                "HOOP_KEY environment variable is not set. This may lead to insecure configurations."
+            ))?;
+        }
+        let path = get_path();
+        let conf = load_conf_file(&path)
+            .unwrap_or_else(|e| panic!("failed to load config file at {path}: {e}"));
+        let conf_file = match conf {
+            Some(c) => ConfFile {
+                token: hoop_key,
+                ..c
+            },
+            None => {
+                Err(anyhow::anyhow!(
+                    "No config file found at {path}. Please create a configuration file or set HOOP_CERT=true to auto-generate certificates."
+                ))?;
+                ConfFile {
+                    hostname: None,
+                    token: hoop_key,
+                    tls_certificate_source: Some(CertSource::External),
+                    tls_certificate_file: None,
+                    tls_private_key_file: None,
+                    tls_verify_strict: Some(true),
+                }
+            }
+        };
```

Comment:
I guess something like this would be better:
```rust
let c = conf.ok_or_else(|| anyhow::anyhow!(
    "No config file found at {path}. Please create a configuration file or set HOOP_CERT=true to auto-generate certificates."
))?;

let conf_file = ConfFile {
    token: hoop_key,
    ..c
};
```